File: sales.admin.web/src/app/salesPermissionKeys.ts
export const SalesPermissionKeys = {
  OrdersView: 'Sales.Orders.View',
  OrdersDetailView: 'Sales.Orders.Detail.View',
  OrdersTimelineView: 'Sales.Orders.Timeline.View',
  OrdersCreate: 'Sales.Orders.Create',
  OrdersShip: 'Sales.Orders.Ship',
  OrdersDeliver: 'Sales.Orders.Deliver',
  OrdersReturn: 'Sales.Orders.Return',
  OrdersRefund: 'Sales.Orders.Refund',
  OrdersRetryPayment: 'Sales.Orders.RetryPayment',
  OrdersCancel: 'Sales.Orders.Cancel',
  OrdersCancelLines: 'Sales.Orders.CancelLines',
  OrdersNotesView: 'Sales.Orders.Notes.View',
  OrdersNotesAdd: 'Sales.Orders.Notes.Add',

  RefundsView: 'Sales.Refunds.View',
  RefundsRequest: 'Sales.Refunds.Request',
  RefundsApprove: 'Sales.Refunds.Approve',
  RefundsReject: 'Sales.Refunds.Reject',
  RefundsComplete: 'Sales.Refunds.Complete',

  ReportsDailyView: 'Sales.Reports.Daily.View',
  ReportsMonthlyView: 'Sales.Reports.Monthly.View',
  ReportsBySiteView: 'Sales.Reports.BySite.View',
  ReportsByProductView: 'Sales.Reports.ByProduct.View',
  ReportsByCustomerView: 'Sales.Reports.ByCustomer.View',
} as const

export type SalesPermissionKey = typeof SalesPermissionKeys[keyof typeof SalesPermissionKeys]

File: Sales.Api/Permissions/SalesPermissionKeys.cs
namespace Sales.Api.Permissions;

public static class SalesPermissionKeys
{
    public const string OrdersView = "Sales.Orders.View";
    public const string OrdersDetailView = "Sales.Orders.Detail.View";
    public const string OrdersTimelineView = "Sales.Orders.Timeline.View";
    public const string OrdersCreate = "Sales.Orders.Create";
    public const string OrdersShip = "Sales.Orders.Ship";
    public const string OrdersDeliver = "Sales.Orders.Deliver";
    public const string OrdersReturn = "Sales.Orders.Return";
    public const string OrdersRefund = "Sales.Orders.Refund";
    public const string OrdersRetryPayment = "Sales.Orders.RetryPayment";
    public const string OrdersCancel = "Sales.Orders.Cancel";
    public const string OrdersCancelLines = "Sales.Orders.CancelLines";
    public const string OrdersNotesView = "Sales.Orders.Notes.View";
    public const string OrdersNotesAdd = "Sales.Orders.Notes.Add";

    public const string RefundsView = "Sales.Refunds.View";
    public const string RefundsRequest = "Sales.Refunds.Request";
    public const string RefundsApprove = "Sales.Refunds.Approve";
    public const string RefundsReject = "Sales.Refunds.Reject";
    public const string RefundsComplete = "Sales.Refunds.Complete";

    public const string ReportsDailyView = "Sales.Reports.Daily.View";
    public const string ReportsMonthlyView = "Sales.Reports.Monthly.View";
    public const string ReportsBySiteView = "Sales.Reports.BySite.View";
    public const string ReportsByProductView = "Sales.Reports.ByProduct.View";
    public const string ReportsByCustomerView = "Sales.Reports.ByCustomer.View";
}
